fabtests/functional/atomic_verify.c

```diff
@@ -0,0 +1,214 @@
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+
+#include <rdma/fi_errno.h>
+#include <rdma/fi_atomic.h>
+
+#include "shared.h"
+#include "hmem.h"
+#include "rdma/fi_domain.h"
+
+int do_verification = 1;
+
+// ofi_atomic doesn't specify the total operation count, just in blocks of WRITE,READWRITE,SWAP.
+// SWAP is last so total count will be SWAP_OP_LAST (which already has the +1 included)
+#define OFI_ALL_OP_CNT FI_ATOMIC_OP_LAST
+
+
+#define ATOM_FOR_FI_MIN(x,y)  (((x) < (y)) ? x : y)
+#define ATOM_FOR_FI_MAX(x,y)  (((x) > (y)) ? x : y)
+#define ATOM_FOR_FI_SUM(x,y)  ((x) + (y))
+#define ATOM_FOR_FI_PROD(x,y) ((x) * (y))
+#define ATOM_FOR_FI_LOR(x,y)  ((x) || (y))
+#define ATOM_FOR_FI_LAND(x,y) ((x) && (y))
+#define ATOM_FOR_FI_BOR(x,y)  ((x) | (y))
+#define ATOM_FOR_FI_BAND(x,y) ((x) & (y))
+#define ATOM_FOR_FI_LXOR(x,y) (((x) && !(y)) || (!(x) && (y)))
+#define ATOM_FOR_FI_BXOR(x,y) ((x) ^ (y))
+
+#define ATOM_CTYPE_FOR_FI_INT32 int32_t
+#define ATOM_CTYPE_FOR_FI_INT16 int16_t
+#define ATOM_CTYPE_FOR_FI_INT8 int8_t
+#define ATOM_CTYPE_FOR_FI_INT64 int64_t
+#define ATOM_CTYPE_FOR_FI_UINT32 uint32_t
+#define ATOM_CTYPE_FOR_FI_UINT16 uint16_t
+#define ATOM_CTYPE_FOR_FI_UINT8 uint8_t
+#define ATOM_CTYPE_FOR_FI_UINT64 uint64_t
+#define ATOM_CTYPE_FOR_FI_FLOAT float
+#define ATOM_CTYPE_FOR_FI_DOUBLE double
+#define ATOM_CTYPE_FOR_FI_LONG_DOUBLE long double
+
+#define atomic_case(ftype, fop) \
+case ftype*OFI_ALL_OP_CNT + fop: \
+    { *(ATOM_CTYPE_FOR_##ftype*)ans = \
+        ATOM_FOR_##fop( \
+            *(ATOM_CTYPE_FOR_##ftype*)val1, \
+            *(ATOM_CTYPE_FOR_##ftype*)val2 ); \
+    } \
+    break;
+
+#define atomic_int_ops(int_type) \
```

Comment:
So far I'm only handling the basic ops.  Fetches and Compare-And-Swap not yet done